File: sdks/php/src/Subscriptions/Requests/CancelSubscriptionDto.php
<?php

namespace NovaBilling\Subscriptions\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Subscriptions\Types\CancelSubscriptionDtoCancelAt;
use NovaBilling\Core\Json\JsonProperty;

class CancelSubscriptionDto extends JsonSerializableType
{
    /**
     * @var value-of<CancelSubscriptionDtoCancelAt> $cancelAt When to cancel: immediately or at end of current period
     */
    #[JsonProperty('cancelAt')]
    public string $cancelAt;

    /**
     * @param array{
     *   cancelAt: value-of<CancelSubscriptionDtoCancelAt>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->cancelAt = $values['cancelAt'];
    }
}

File: sdks/csharp/src/NovaBilling/Subscriptions/Requests/CancelSubscriptionDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record CancelSubscriptionDto
{
    /// <summary>
    /// Subscription ID
    /// </summary>
    [JsonIgnore]
    public required string Id { get; set; }

    /// <summary>
    /// When to cancel: immediately or at end of current period
    /// </summary>
    [JsonPropertyName("cancelAt")]
    public required CancelSubscriptionDtoCancelAt CancelAt { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
